### PR TITLE
chore: internalise `ethers` utilities for `hexlify`, `arrayify`, `concat` and `BytesLike`

### PR DESCRIPTION
**Pull request by danielbate**
Following on from discussions in #1592 to deprecate `ethers` functionality and improve tree-shakability, this PR removes usages of the following `ethers` functions:

 - [x] `hexlify`
 - [x] `arrayify`
 - [x] `concat`
 - [x] `BytesLike`

### CODE REVIEW

File: packages/crypto/tsup.config.bundled_xclud6u4tvq.mjs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
...bWVudHMvRGV2ZWxvcG1lbnQvRnVlbC90cy9wYWNrYWdlcy9jcnlwdG8vdHN1cC5jb25maWcudHNcIjtjb25zdCBfX2luamVjdGVkX2Rpcm5hbWVfXyA9IFwiL1VzZXJzL2RhbmllbC9Eb2N1bWVudHMvRGV2ZWxvcG1lbnQvRnVlbC90cy9wYWNrYWdlcy9jcnlwdG9cIjtjb25zdCBfX2luamVjdGVkX2ltcG9ydF9tZXRhX3VybF9fID0gXCJmaWxlOi8vL1VzZXJzL2RhbmllbC9Eb2N1bWVudHMvRGV2ZWxvcG1lbnQvRnVlbC90cy9wYWNrYWdlcy9jcnlwdG8vdHN1cC5jb25maWcudHNcIjtpbXBvcnQgeyBpbmRleCB9IGZyb20gJ0BpbnRlcm5hbC90c3VwJztcbmltcG9ydCB0eXBlIHsgT3B0aW9ucyB9IGZyb20gJ3RzdXAnO1xuXG5leHBvcnQgY29uc3Qga2V5c3RvcmVPcHRpb25zOiBPcHRpb25zW10gPSBbXG4gIHtcbiAgICAuLi5pbmRleCxcbiAgICBlbnRyeTogWydzcmMvaW5kZXgudHMnXSxcbiAgICBmb3JtYXQ6IFsnY2pzJ10sXG4gIH0sXG4gIHtcbiAgICAuLi5pbmRleCxcbiAgICBlbnRyeTogWydzcmMvaW5kZXgudHMnLCAnc3JjL2luZGV4LmJyb3dzZXIudHMnXSxcbiAgICBmb3JtYXQ6IFsnZXNtJ10sXG4gIH0sXG5dO1xuXG5leHBvcnQgZGVmYXVsdCBrZXlzdG9yZU9wdGlvbnM7XG4iXSwKICAibWFwcGluZ3MiOiAiO0FBQStULFNBQVMsYUFBYTtBQUc5VSxJQUFNLGtCQUE2QjtBQUFBLEVBQ3hDO0FBQUEsSUFDRSxHQUFHO0FBQUEsSUFDSCxPQUFPLENBQUMsY0FBYztBQUFBLElBQ3RCLFFBQVEsQ0FBQyxLQUFLO0FBQUEsRUFDaEI7QUFBQSxFQUNBO0FBQUEsSUFDRSxHQUFHO0FBQUEsSUFDSCxPQUFPLENBQUMsZ0JBQWdCLHNCQUFzQjtBQUFBLElBQzlDLFFBQVEsQ0FBQyxLQUFLO0FBQUEsRUFDaEI7QUFDRjtBQUVBLElBQU8sc0JBQVE7IiwKICAibmFtZXMiOiBbXQp9Cg==
```
Comment (arboleya) [inline packages/crypto/tsup.config.bundled_xclud6u4tvq.mjs:20]: Was this file added by mistake?

Comment (danielbate) [reply packages/crypto/tsup.config.bundled_xclud6u4tvq.mjs:20]: Indeed

Comment (danielbate) [reply packages/crypto/tsup.config.bundled_xclud6u4tvq.mjs]: [b1d6458](https://github.com/FuelLabs/fuels-ts/pull/1775/commits/b1d645804d2d33b801a3d743a9dd90ff89d11224)

**Review by arboleya — APPROVED** [review]

File: packages/crypto/tsup.config.bundled_xclud6u4tvq.mjs (hunk 1 above)
Comment (danielbate) [reply packages/crypto/tsup.config.bundled_xclud6u4tvq.mjs]: #1779

**Review by Dhaiwat10 — DISMISSED** [review]

**Review by arboleya — DISMISSED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**